### PR TITLE
reportException instead of canceling tasks with bad payloads

### PR DESCRIPTION
**Pull request by bhearsum**

### CODE REVIEW

File: bbb/services.py [hunk 1]
```diff
@@ -402,7 +404,11 @@ def handlePending(self, data, msg):
             # malformed-payload is the most accurate TC status for this situation
             # but we can't use reportException for cancelling pending tasks,
             # so this will show up as "cancelled" on TC.
```
Comment (jonasfj) [inline bbb/services.py]: Lol, maybe update the comment :)

**Comment by jonasfj** [comment]
`r+`...

Note: I would normally structure the code as:
 1. Get message from pulse
 2. claimTask
 3. Ack message from pulse (queue will retry if you fail after calling claimTask)
 4. download task definition
 5. validation task definition (report malformed-payload and finish if needed)
 6. run task while reclaiming (inject into buildbot)

You structure works too... and I doubt it's worth refactoring for.

**Comment by bhearsum** [comment]
That's a good point. I can't quite do all of it, because we're only claiming in handlePending if we're going to cancel (otherwise we wait until the Buildbot Build starts, so that the "running" time in TC matches Buildbot). However, I should ack the message immediately after claiming. I've been trying to do message acking immediately after the first unretriable operation, I just forgot to update that here :).

**Comment by bhearsum** [comment]
One thing I just thought of here is

**Pull request merged**